privacyidea/lib/eventhandler/webhookevanthandler.py

```diff
@@ -0,0 +1,150 @@
+    """
+    With the WebHook Handler, a webhook can be sent at a particular event
+    """
+
+    identifier = "WebHookHandler"
+    description = "This eventhandler can post webhooks"
+
+    @property
+    def allowed_positions(cls):
+        """
+        This returns the allowed positions of the event handler definition.
+
+        :return: list of allowed positions
+        """
+        return ["post", "pre"]
+
+    @property
+    def actions(cls):
+        """
+        This method returns a dictionary of allowed actions and possible
+        options in this handler module.
+
+        :return: dict with actions
+        """
+        # The event handler has just one action. Maybe we can  hide action select for the clarity of the UI
+        actions = {ACTION_TYPE.POST_WEBHOOK: {
+            "URL": {
+                "type": "str",
+                "required": True,
+                "description": _("The URL the WebHook is posted to")
+            },
+            "content_type": {
+                "type": "str",
+                "required": True,
+                "description": _("The encoding that is sent to the WebHook, for example json"),
+                "value": [
+                    CONTENT_TYPE.JSON,
+                    CONTENT_TYPE.URLENCODED]
+            },
+            "data": {
+                "type": "str",
+                "required": True,
+                "description": _("The data posted in the WebHook")
+            }
+        }}
+        return actions
+
+    def do(self, action, options=None):
+        """
+        This method executes the defined action in the given event.
+
+        :param action: The action to perform
+        :type action: str
+        :param options: Contains the flask parameters g, URL, data, parameter and the fire and forget
+        :type options: dict
+        :return:
+        """
+        ret = True
+        g = options.get("g")
+        request = options.get("request")
+        handler_def = options.get("handler_def")
+        handler_options = handler_def.get("options")
+        webhook_url = handler_options.get("URL")
+        webhook_text = handler_options.get("data")
+        content_type = handler_options.get("content_type")
+
+        # Replace placeholder in data
+        logged_in_user = g.logged_in_user.get("username")
+        attributes = {
+            'logged_in_user': logged_in_user
+        }
+        webhook_text = webhook_text.format_map(defaultdict(str, attributes))
+
+        if action.lower() == ACTION_TYPE.POST_WEBHOOK:
+            try:
+                if content_type in (CONTENT_TYPE.URLENCODED, CONTENT_TYPE.JSON):
+                    """
+                    This is the main function where the webhook is sent.
+                    Documentation of the requests function is fond her docs.python-requests.org
+                    """
+                    resp = requests.post(webhook_url, data=json.dumps(webhook_text, sort_keys=True, default=str),
```

Comment:
I am not sure about how the `json.dumps` would work out.

Would you always to `json.dumps` or only, when you choose 'CONTENT_TPYE.JSON'?